mathesar_ui/src/systems/table-view/table-inspector/column/ColumnType.svelte

```diff
@@ -1,62 +1,30 @@
 <script lang="ts">
-  import { Button, Icon } from '@mathesar/component-library';
-  import { iconEdit } from '@mathesar/icons';
   import {
     getTabularDataStoreFromContext,
     type ProcessedColumn,
   } from '@mathesar/stores/table-data';
-  import { MissingExhaustiveConditionError } from '@mathesar/utils/errors';
   import { AbstractTypeControl } from '@mathesar/components/abstract-type-control';
   import type { ColumnTypeOptionsSaveArgs } from '@mathesar/components/abstract-type-control/types';
-  import { toast } from '@mathesar/stores/toast';
 
   const tabularData = getTabularDataStoreFromContext();
   $: ({ columnsDataStore } = $tabularData);
 
   export let column: ProcessedColumn;
 
-  let mode: 'read' | 'edit' = 'read';
-  function toggleMode(): undefined {
-    switch (mode) {
-      case 'read':
-        mode = 'edit';
-        break;
-      case 'edit':
-        mode = 'read';
-        break;
-      default:
-        throw new MissingExhaustiveConditionError(mode, 'ColumnType');
-    }
-    return undefined;
-  }
-
-  async function save(columnInfo: ColumnTypeOptionsSaveArgs) {
-    try {
-      await columnsDataStore.patch(column.id, {
-        type: columnInfo.type,
-        type_options: columnInfo.type_options,
-        display_options: columnInfo.display_options,
-      });
-    } catch (err) {
-      const message =
-        err instanceof Error ? err.message : 'Unable to change column type.';
-      toast.error(message);
-    }
+  async function save(
+    columnInfo: Pick<ColumnTypeOptionsSaveArgs, 'type' | 'type_options'>,
+  ) {
+    await columnsDataStore.patch(column.id, {
+      type: columnInfo.type,
+      type_options: columnInfo.type_options,
```

Comment:
Type changes should change/reset default_value and display_options.

Reply:
61a725d8